docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+- the UI controls in use by the block will be automatically updated as they are improved, without the block author having to do anything
+
+#### Current limits of the Block Supports API
+
+While the Block Supports API provides value, it also comes with some limitations a block author needs to be aware of. To better visualize what they are, let's run with the following example of a table block:
+
+```html
+<table>
+  <thead><tr><th>Header</th></tr></thead>
+  <tbody>
+    <tr><th>First</th></tr>
+    <tr><th>Second</th></tr>
+  </tbody>
+  <tfoot>
+    <tr><th>Footer</th></tr>
+  </tfoot>
+</table>
+```
+
+1. **Only one style type per block.**
+
+One of the limitations is that, from all the [styles available](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/), there can be only one instance of any them in use by the block. Following the example, the table block can only have a single font size. If the block author wanted to have three different font sizes (head, body, and footer) it can't do it using the current block supports API. See [this issue](https://github.com/WordPress/gutenberg/issues/33255) for more detailed info and ways forward.
+
+2. **Styles are serialized to the outermost HTML node of the block, the wrapper.**
+
+The Block Supports API only serializes the font size value to the wrapper, resulting in the following HTML `<table class="has-small-font-size">` . The current block supports API doesn't serialize this value to a different node, for example, the `<tbody>`.
+
+This is an active area of work you can follow [in the tracking issue](https://github.com/WordPress/gutenberg/issues/38167). The linked proposal is exploring a different way to serialize the user changes: instead of each block support serializing its own data (for example, classes such as `has-small-font-size`, `has-green-color`) the idea is the block would get a single class instead (for example, `wp-style-UUID`) and the contents of that class will be generated in the server by WordPress.
+
+While work continues in that proposal, there's a scape hatch, an experimental option block authors can use. Any block support can skip the serialization to HTML markup by using  `__experimentalSkipSerialization`. For example:
+
+```
+{
+  "name": "core/paragraph", 
+  "...": "...",
+  "supports": {
+	"typography": {
+		"fontSize": true,
+		"__experimentalSkipSerialization": true
+	}
+  }
+}
+```
+
+This means that the font size block supports will do all of the things (create an UI control, bind the block attribute to the control, etc) except serializing the user values into the HTML markup. The classes and inline styles will not be automatically applied to the wrapper and is the block author responsibility to implement this in the `edit`, `save`, and `render_callback` functions. See [this issue](https://github.com/WordPress/gutenberg/issues/28913) for examples of how it was done for some blocks provided by WordPress.
```

Comment:
The `__experimentalSkipSerialization` here occurs at the block support level (`typography`) rather than the individual feature within that block support, so I've added a suggestion:

```suggestion
This means that the typography block support will do all of the things (create a UI control, bind the block attribute to the control, etc) except serializing the user values into the HTML markup. The classes and inline styles will not be automatically applied to the wrapper and it is the block author's responsibility to implement this in the `edit`, `save`, and `render_callback` functions. See [this issue](https://github.com/WordPress/gutenberg/issues/28913) for examples of how it was done for some blocks provided by WordPress.
```

Reply:
There is currently an open PR to [allow skipping serialization of individual block support features.](https://github.com/WordPress/gutenberg/pull/36293). It shouldn't be too far off landing 🤞

Reply:
Oh, thanks @aaronrobertshaw, I totally forgot about that one! 🙇

Reply:
https://github.com/WordPress/gutenberg/pull/38208/commits/fd395d425fdfa2a1b8fca7bf836cd3198c32f1b8